web-server/app/codesearch/web/controllers/SearchController.scala

```diff
@@ -36,16 +36,21 @@ trait SearchController[V <: DefaultTable] { self: InjectedController =>
   }
 
   def search(query: String,
-             filter: String,
+             filter: Option[String],
              caseInsensitive: String,
              spaceInsenstive: String,
              precise: String,
              sources: String,
              page: String): Action[AnyContent] =
     Action.async { implicit request =>
-      val request = SearchRequest.applyRaw(query, filter, caseInsensitive, spaceInsenstive, precise, sources, page)
-      val callURI =
-        s"/$lang/search?query=$query?filter=$filter&caseInsensitive=$caseInsensitive&spaceInsensitive=$spaceInsenstive&precise=$precise&sources=$sources"
+      val request =
+        SearchRequest.applyRaw(query, filter, caseInsensitive, spaceInsenstive, precise, sources, page)
+
+      val callURI = filter match {
```

Comment:
You can make the `callURI ` function a member of the `SearchRequest` case class

Reply:
I did, but I'm not sure if it is https://github.com/aelve/codesearch/pull/192/commits/6e4c56b5c3046930e607876f7887872685db6b12

Reply:
@kamilongus I think that it was not done in the best way, you just said it, but I think you meant something else.